syntax = "proto3";

package hybrtc;

// uint32 verifier_id  = 16; // 10
// bytes verifier_pk   = 17; // 11
// uint32 attester_id  = 32; // 20
// bytes attester_pk   = 33; // 21
// bytes format_settings   = 48; // 30
// bytes evidence      = 49; // 31

message AttestationChallenge {
    uint32 verifier_id  = 16; // 10
    bytes verifier_pk   = 17; // 11
    bytes format_settings   = 48; // 30
}

message AttestationResponse {
    uint32 verifier_id  = 16; // 10
    uint32 attester_id  = 32; // 20
    bytes attester_pk   = 33; // 21
    bytes evidence      = 49; // 31
}

message Pair {
    bytes key = 1;
    bytes value = 2;
}

message Pairs {
    repeated Pair pairs = 1;
}

message QueryRequest {
    uint32 server_id = 1;
    uint32 server_count = 2;
    bytes homo_pk = 3;
}

message QueryResponse {
    repeated Pair pairs = 1;
}

message ComputeRequest {
    uint32 initiator_id = 1;
    uint32 sender_id = 2;
    bytes bloom_filter = 10;
}

message ComputeResponse {
    uint32 initiator_id = 1;
    uint32 sender_id = 2;
    repeated Pair pairs = 10;
}
